crates/commonware-node/src/db/mod.rs

```diff
@@ -0,0 +1,571 @@
+                None => Ok(None), // Marked for deletion
+            };
+        }
+
+        self.db.get(key).await
+    }
+
+    /// Insert a key-value pair into the transaction's pending writes.
+    ///
+    /// The value is stored in its typed form and only serialized at commit time.
+    /// This does not immediately write to the store - call `commit()` to persist.
+    pub fn insert<K, V>(&mut self, key: K, value: V)
+    where
+        K: AsRef<[u8]>,
+        V: CodecWrite + EncodeSize + Send + Sync + 'static,
+    {
+        let key_hash = key_to_b256(key.as_ref());
+        self.writes
+            .insert(key_hash, Some(Box::new(value) as Box<dyn CachedValue>));
+    }
+
+    /// Remove a key from the store.
+    ///
+    /// This marks the key for deletion. The actual removal happens when `commit()` is called.
+    pub fn remove<K>(&mut self, key: K)
+    where
+        K: AsRef<[u8]>,
+    {
+        let key_hash = key_to_b256(key.as_ref());
+        self.writes.insert(key_hash, None);
+    }
+
+    /// Check if a key exists in the store.
+    ///
+    /// This checks pending writes first, then falls back to the database.
+    pub async fn exists<K>(&self, key: K) -> bool
+    where
+        K: AsRef<[u8]>,
+    {
+        let key_hash = key_to_b256(key.as_ref());
+
+        if let Some(value_opt) = self.writes.get(&key_hash) {
+            return value_opt.is_some();
+        }
+
+        self.db.get_raw(key).await.is_some()
+    }
+
+    /// Get the node version that last wrote to this database.
+    ///
+    /// Returns None if no version has been written yet (new database).
+    pub async fn get_node_version(&self) -> Result<Option<String>, eyre::Error> {
+        let key_hash = key_to_b256(NODE_VERSION_KEY.as_bytes());
+
+        // Check pending writes first
+        if let Some(value_opt) = self.writes.get(&key_hash) {
+            return match value_opt {
+                Some(cached) => Ok(Some(
+                    String::from_utf8(cached.serialize()?.to_vec())
+                        .map_err(|e| eyre::eyre!("invalid utf8: {}", e))?,
+                )),
+                None => Ok(None),
+            };
+        }
+
+        // Get raw bytes from storage (node version bypasses typed cache)
+        Ok(self
+            .db
+            .get_raw(NODE_VERSION_KEY)
+            .await
+            .and_then(|b| String::from_utf8(b.to_vec()).ok()))
+    }
+
+    /// Set the current node version in the database.
+    ///
+    /// This should be called when the database schema is initialized or migrated
+    /// to track which node version last modified the database.
+    pub fn set_node_version(&mut self, version: String) {
+        self.insert(NODE_VERSION_KEY, version.into_bytes())
+    }
+
+    // ── Ceremony Store ──────────────────────────────────────────────────────
+
+    /// Get ceremony state for a specific epoch.
+    pub async fn get_ceremony(&self, epoch: u64) -> Result<Option<ceremony::State>, eyre::Error> {
```

Comment:
Do these need to be on the low level tx? Can we make this a new-type wrapper that is specific to the DKG actor? I.e. `crate::dkg::manager::Tx(crate::db::Tx).`

I like that that the DKG actor has its own database where it need not worry about anybody else messing with it.